contracts/rewarders/SimpleRewarderPerSec.sol

```diff
@@ -108,6 +113,13 @@ contract SimpleRewarderPerSec is IRewarder, BoringOwnable, ReentrancyGuard {
         MCJ = _MCJ;
         isNative = _isNative;
         poolInfo = PoolInfo({lastRewardTimestamp: block.timestamp, accTokenPerShare: 0});
+
+        JoePair pair = JoePair(address(_lpToken));
+        IERC20Metadata token0 = IERC20Metadata(pair.token0());
+        IERC20Metadata token1 = IERC20Metadata(pair.token1());
+
+        if (_tokenPerSec > 10**6) ACC_TOKEN_PRECISION = 10**(36 - (token0.decimals() + token1.decimals()) / 2);
```

Comment:
Don't forget to do this check/calculation again in `setRewardRate`

Reply:
Really good catch, this actually involves some changes on `accTokenPerShare` too !